armi/nuclearDataIO/cccc/dif3d.py

```diff
@@ -0,0 +1,237 @@
+        k-effective).
+
+        The parameters are stored as a dictionary under the attribute `threeD`.
+
+        Parameters
+        ----------
+        None
+        """
+        with self.createRecord() as record:
+            for param in FILE_SPEC_3D_PARAMS:
+                self._data.threeD[param] = record.rwDouble(self._data.threeD[param])
+
+    def _rw4DRecord(self) -> None:
+        """
+        Record for the optimsum overrelaxation factors. This record is only present when
+        using DIF3D-FD and if `NUMORP` is greater than 0.
+
+        The parameters are stored as a dictionary under the attribute `fourD`. This
+        could be changed into a list in the future since this record represents groupwise
+        data.
+
+        Parameters
+        ----------
+        None
+        """
+        if self._data.twoD["NUMORP"] != 0:
+            omegaParams = [f"OMEGA{e}" for e in range(1, self._data.twoD["NUMORP"] + 1)]
+
+            with self.createRecord() as record:
+                # Initialize the record if we're reading
+                if self._data.fourD is None:
+                    self._data.fourD = {omegaParam: None for omegaParam in omegaParams}
+
+                for omegaParam in omegaParams:
+                    self._data.fourD[omegaParam] = record.rwDouble(
+                        self._data.fourD[omegaParam]
+                    )
+
+    def _rw5DRecord(self) -> None:
+        """
+        Record for the axial coarse-mesh rebalance boundaries. Coarse mesh balancing is
+        disabled in DIF3D-VARIANT, so this record is only relevant for DIF3D-Nodal. This
+        record is only present if `NCMRZS` is greater than 0.
+
+        The parameters are stored as a dictionary under the attribute `fiveD`.
+
+        Parameters
+        ----------
+        None
+        """
+        if self._data.twoD["NCMRZS"] != 0:
+            zcmrcParams = [f"ZCMRC{e}" for e in range(1, self._data.twoD["NCMRZS"] + 1)]
+            nzintsParams = [
+                f"NZINTS{e}" for e in range(1, self._data.twoD["NCMRZS"] + 1)
+            ]
+
+            with self.createRecord() as record:
+                # Initialize the record if we're reading
+                if self._data.fiveD is None:
+                    self._data.fiveD = {zcmrcParam: None for zcmrcParam in zcmrcParams}
+                    self._data.fiveD.update(
+                        {nzintsParam: None for nzintsParam in nzintsParams}
+                    )
+
+                for zcmrcParam in zcmrcParams:
+                    self._data.fiveD[zcmrcParam] = record.rwDouble(
+                        self._data.fiveD[zcmrcParam]
+                    )
+                for nzintsParam in nzintsParams:
+                    self._data.fiveD[nzintsParam] = record.rwInt(
+                        self._data.fiveD[nzintsParam]
+                    )
+
+    def readWrite(self):
+        """"""
```

Comment:
Empty docsting.  Put a sentence in here, or just delete the quotes. IMO

Reply:
oops that's because I saw a squirrel. Fixing.